pollon.go

```diff
@@ -17,61 +17,125 @@ package pollon
 import (
 	"fmt"
 	"io"
+	"math/rand"
 	"net"
 	"sync"
+	"sync/atomic"
 	"time"
 )
 
+type LBType string
+
+const (
+	Random     LBType = "random"
+	LeastQueue LBType = "leastqueue"
+)
+
 type ConfData struct {
-	DestAddr *net.TCPAddr
+	DestAddr []*net.TCPAddr
 }
 
-type Proxy struct {
-	C          chan ConfData
-	listener   *net.TCPListener
-	confMutex  sync.Mutex
+type Backend struct {
 	destAddr   *net.TCPAddr
 	closeConns chan struct{}
-	stop       chan struct{}
-	endCh      chan error
-	connMutex  sync.Mutex
+	needClean  bool
+	connNum    int32
+}
 
+type Proxy struct {
+	C                 chan ConfData
+	listener          *net.TCPListener
+	confMutex         sync.Mutex
+	connMutex         sync.Mutex
+	stop              chan struct{}
+	endCh             chan error
 	keepAlive         bool
 	keepAliveIdle     time.Duration
 	keepAliveCount    int
 	keepAliveInterval time.Duration
+	backends          []*Backend
+	lbType            LBType
 }
 
 func NewProxy(listener *net.TCPListener) (*Proxy, error) {
 	return &Proxy{
-		C:          make(chan ConfData),
-		listener:   listener,
-		closeConns: make(chan struct{}),
-		stop:       make(chan struct{}),
-		endCh:      make(chan error),
-		connMutex:  sync.Mutex{},
+		C:        make(chan ConfData),
+		listener: listener,
+		stop:     make(chan struct{}),
+		endCh:    make(chan error),
+		lbType:   Random,
 	}, nil
 }
 
+func newBackend(destAddr *net.TCPAddr) *Backend {
+	return &Backend{
+		destAddr:   destAddr,
+		closeConns: make(chan struct{}),
+	}
+}
+
+func (p *Proxy) GetBackend() *Backend {
+	if p.lbType == LeastQueue {
+		var backResult *Backend = nil
+		var connNum int32
+		for _, b := range p.backends {
+			if backResult == nil {
+				backResult = b
+				connNum = atomic.LoadInt32(&b.connNum)
+				continue
```

Comment:
not a biggie.

Maybe rename it as leastConnNum or something to signify that's the least connection count.